server/src/main/java/org/elasticsearch/search/sort/SortFieldAndFormat.java

```diff
@@ -13,10 +13,7 @@
 
 import java.util.Objects;
 
-public final class SortFieldAndFormat {
-
-    public final SortField field;
-    public final DocValueFormat format;
+public record SortFieldAndFormat(SortField field, DocValueFormat format) {
```

Comment:
shall we extract making this a record from this PR? Otherwise it's hard to follow what else you are changing in the same PR.

Reply:
Right fair point let me do that real quick :)

Reply:
There we go :) https://github.com/elastic/elasticsearch/pull/113398